Lib/test/test_sqlite3/test_dump.py

```diff
@@ -190,6 +190,21 @@ def __getitem__(self, index):
         got = list(self.cx.iterdump())
         self.assertEqual(expected, got)
 
+    def test_dump_custom_row_factory(self):
+        # gh-118221: iterdump should be able to cope with custom row factories
+        def dict_factory(cu, row):
+            fields = [col[0] for col in cu.description]
+            return {k: v for k, v in zip(fields, row)}
+
+        self.cx.row_factory = dict_factory
+        CREATE_TABLE = "CREATE TABLE test(t);"
+        expected = ["BEGIN TRANSACTION;", CREATE_TABLE, "COMMIT;"]
+
+        self.cu.execute(CREATE_TABLE)
```

Comment:
Should not we create a new cursor after changing connection's `row_factory`? :thinking:

Reply:
It does not matter for the test since `iterdump()` creates a new cursor anyway. Also, `CREATE TABLE` does not return any rows :)

Reply:
Right, thanks  :+1: